torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1165,6 +1165,40 @@ void LoopNest::reorderAxis(For* a, For* b) {
   }
 } // namespace tensorexpr
 
+void LoopNest::unroll(For* f, Stmt** unrolled) {
+  Block* p = dynamic_cast<Block*>(f->get_parent());
+  if (!f) {
+    throw malformed_input("unroll attempted on null loop", f);
+  } else if (!p) {
+    throw malformed_input("unroll attempted on loop with no parent", p);
+  }
+
+  const auto start = dynamic_cast<const IntImm*>(f->start());
+  const auto stop = dynamic_cast<const IntImm*>(f->stop());
+  if (!start) {
+    throw std::runtime_error("Can't unroll due to non-constant loop start!");
+  }
+  if (!stop) {
+    throw std::runtime_error("Can't unroll due to non-constant loop stop!");
+  }
+
+  std::vector<Stmt*> unrolled_stmts;
+  int start_val = start->value();
+  int stop_val = stop->value();
+  for (int current = start_val; current < stop_val; ++current) {
+    for (const auto stmt : f->body()->stmts()) {
+      auto stmt_copy = Stmt::clone(stmt);
+      unrolled_stmts.push_back(
+          Substitute(stmt_copy, {{f->var(), new IntImm(current)}}));
+    }
+  }
+  *unrolled = new Block(unrolled_stmts);
+  *unrolled = IRSimplifier::simplify(*unrolled);
+
+  // TODO: cleanup API for adding/removing statements
```

Comment:
Is this comment relevant or did it come from splitWithTail?

Reply:
It came from `splitWithTail` - I thought if it was relevant there, it'd be relevant here too. Should I remove it?

Reply:
I'm not sure exactly what it's referring to, it's probably not relevant.

Reply:
Removed.